tests/check_framework/test_settings.py

```diff
@@ -0,0 +1,22 @@
+from django.apps import apps
+from django.core import checks
+from django.test import SimpleTestCase, override_settings
+
+
+class SettingsDeprecationCheckTests(SimpleTestCase):
+    @override_settings(TRANSACTIONS_MANAGED=True)
+    def test_check_deprecated_settings(self):
+        all_issues = checks.run_checks(app_configs=apps.get_app_configs())
+
+        self.assertGreater(len(all_issues), 0)
+
+        self.assertIn(
+            checks.Warning(
+                "You still use 'TRANSACTIONS_MANAGED' in your Django settings "
+                "file. This attribute is deprecated.",
+                hint="Please refer to the documentation and remove/replace "
+                "this attribute.",
+                id="settings.W001",
```

Comment:
Checks need to be documented in `docs/ref/checks.txt`

Reply:
@sarahboyce Thx for the hint! Wrote a two-liner. Will this suffice?

Reply:
> WARNING: role 'cve' is already registered, it will be overridden [app.add_role]

I get this error in the pipeline. Any idea what I can do about it? 😔

Reply:
It's unrelated and there was a PR to fix this recently :+1: 

In the section's I see we have a `Backwards compatibility` checks around upgrading Django
I wonder if this check should be documented there :thinking:

Reply:
Hm, this goes in a simlar direction. ([Link](https://docs.djangoproject.com/en/5.1/ref/checks/#backwards-compatibility))

What do you suggest? We move this stuff to the existing check?

Reply:
On the other hand, these checks are targeting the semantics. We just check for existance. That's a difference. Not sure if this is a solid argument, though.

Reply:
I moved the deprecated settings docs next to the backwards compat docs. @sarahboyce Are you ok with this? 🙂

Reply:
@GitRon Besides my comments above about how the check string has to be identical in the checks docs, we should test every single entry in the `DEPRECATED_SETTINGS` set (using `subTest`).

Reply:
That's interesting. I understand that this will cover literally every case but the code is the same. Don't you think this will suffice? If not, I'm happy to add the subTests - but how would you do that? We need to put the variables in the settings to trigger the warning. I went with a override_settings decorator. If I stick to this pattern, I'd have to add one test case (not a subtest) per variable. On secoond thought, this might be exactly what you want to achieve 🤔 Explicit test cases to ensure that everything works as expected?